### PR TITLE
Activated the local software configs feature

### PR DESCRIPTION
**Pull request by dowgird**
TESTED: Ran the locally-built osconfig on a VM, verified that a recipe from metadata is being applied.

### CODE REVIEW

File: policies/policies.go [hunk 1]
```diff
@@ -35,23 +35,33 @@ import (
...
+
 	client, err := agentendpoint.NewClient(ctx)
 	if err != nil {
 		logger.Errorf("agentendpoint.NewClient Error: %v", err)
-		return
+	} else {
+		defer client.Close()
+		resp, err = client.LookupEffectiveGuestPolicies(ctx)
+		if err != nil {
+			logger.Errorf("Error running GuestPolicies: %v", err)
```
Comment (adjackura) [inline policies/policies.go]: Error running LookupEffectiveGuestPolicies

File: policies/policies.go [hunk 2]
```diff
@@ -35,23 +35,33 @@ import (
...
+	local, err := readLocalConfig()
 	if err != nil {
-		logger.Errorf("Error running GuestPolicies: %v", err)
-		return
+		logger.Errorf("Error reading local software config: %v", err)
+		empty := localConfig{}
+		local = &empty
 	}
 
+	effectiveResp := mergeConfigs(local, *resp)
```
Comment (adjackura) [inline policies/policies.go]: We should change mergeConfigs to take *agentendpointpb.EffectiveGuestPolicy

File: policies/policies.go [hunk 3]
```diff
@@ -35,23 +35,33 @@ import (
...
 	}
-	defer client.Close()
 
-	resp, err := client.LookupEffectiveGuestPolicies(ctx)
+	local, err := readLocalConfig()
 	if err != nil {
-		logger.Errorf("Error running GuestPolicies: %v", err)
-		return
+		logger.Errorf("Error reading local software config: %v", err)
+		empty := localConfig{}
```
Comment (adjackura) [inline policies/policies.go]: Same as above, mergeConfigs should just check for a nil localConfig

File: policies/policies.go [hunk 4]
```diff
@@ -35,23 +35,33 @@ import (
...
 	client, err := agentendpoint.NewClient(ctx)
 	if err != nil {
 		logger.Errorf("agentendpoint.NewClient Error: %v", err)
-		return
+	} else {
+		defer client.Close()
+		resp, err = client.LookupEffectiveGuestPolicies(ctx)
+		if err != nil {
+			logger.Errorf("Error running GuestPolicies: %v", err)
+			resp = &agentendpointpb.EffectiveGuestPolicy{}
```
Comment (adjackura) [inline policies/policies.go]: Just have mergeConfigs check for a nil EffectiveGuestPolicy so we don't need to do this here

**Comment by dowgird** [comment]
Thanks for the quick review!

/assign ajackura

**Comment by dowgird** [comment]
Thanks for the quick review!

/assign adjackura

File: policies/local.go [hunk 5]
```diff
@@ -86,33 +95,38 @@ func getID(repo agentendpointpb.PackageRepository) string {
 	}
 }
 
-// MergeConfigs merges the local config with the lookup response, giving priority to the lookup
-// result.
-func mergeConfigs(local *localConfig, egp agentendpointpb.EffectiveGuestPolicy) agentendpointpb.EffectiveGuestPolicy {
+// mergeConfigs merges the local config with the lookup response, giving priority to the lookup
+// result. If both arguments are nil, returns an empty policy.
+func mergeConfigs(local *localConfig, egp *agentendpointpb.EffectiveGuestPolicy) agentendpointpb.EffectiveGuestPolicy {
+	var merged agentendpointpb.EffectiveGuestPolicy
```
Comment (adjackura) [inline policies/local.go]: Just use egp for the return, so edit it in place and return the pointer.
if egp == nil {
egp = &agentendpointpb.EffectiveGuestPolicy{}
}

Comment (dowgird) [reply policies/local.go]: Done.

**Pull request merged**